lib/osf-components/addon/components/form-controls/power-select/component.ts

```diff
@@ -0,0 +1,39 @@
+import { tagName } from '@ember-decorators/component';
+import Component from '@ember/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { ChangesetDef } from 'ember-changeset/types';
+
+import { layout } from 'ember-osf-web/decorators/component';
+import defaultTo from 'ember-osf-web/utils/default-to';
+
+import styles from './styles';
+import template from './template';
+
+@tagName('')
+@layout(template, styles)
+export default class FormControlPowerSelect extends Component {
+    // Required params
+    options!: string[];
+    valuePath!: string;
+    changeset!: ChangesetDef;
+
+    // Optional params
+    shouldShowMessages?: boolean;
+    disabled: boolean = defaultTo(this.disabled, false);
+    onchange?: (option: string) => void;
+
+    didReceiveAttrs() {
+        assert('FormControls::Select - @options are required', Boolean(this.options));
+        assert('FormControls::Select - @valuePath is required', Boolean(this.valuePath));
+        assert('FormControls::Select - @changeset is required', Boolean(this.changeset));
+    }
+
+    @action
+    updateChangeset(option: any) {
```

Comment:
Should the type of `option` be a string?

Reply:
It doesn't have to be. It could by anything as options could also be an Object. I'll also change the `options` param's type to `any`.